### PR TITLE
Pluggable

### PR DESCRIPTION
**Pull request by jdorando**

### CODE REVIEW

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/rest/ManagedFeatureStore.java [hunk 1]
```diff
@@ -77,33 +83,28 @@ protected void onManagedDataLoadedFromStorage(NamedList<?> managedInitArgs,
...
       for (final Map<String,Object> u : up) {
-        update(u);
+        addFeature(u);
       }
     }
   }
 
-  public void update(Map<String,Object> map) {
-    try {
+  public void addFeature(Map<String,Object> map) {
```
Comment (mnilsson23) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/rest/ManagedFeatureStore.java]: Do we need 2 addFeature functions, or can we just have 1?

File: solr/contrib/ltr/src/test/org/apache/solr/ltr/rest/TestModelManager.java [hunk 2]
```diff
@@ -108,7 +108,7 @@ public void testRestManagerEndpoints() throws Exception {
     // Add bad feature (wrong params)_
     final String badfeature = "{\"name\": \"fvalue\", \"class\": \"org.apache.solr.ltr.feature.impl.FieldValueFeature\", \"params\": {\"value\": 1} }";
     assertJPut(CommonLTRParams.FEATURE_STORE_END_POINT, badfeature,
-        "/responseHeader/status==400");
+        "/responseHeader/status==500");
```
Comment (mnilsson23) [inline solr/contrib/ltr/src/test/org/apache/solr/ltr/rest/TestModelManager.java]: Shouldn't this still be a 400 error (bad request)?

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/rest/ManagedModelStore.java [hunk 3]
```diff
@@ -254,7 +261,7 @@ protected Object applyUpdatesToManagedData(Object updates) {
       }
     }
 
-    return store.modelAsManagedResources();
+    return modelAsManagedResources(store);
```
Comment (mnilsson23) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/rest/ManagedModelStore.java]: To be consistent between the Managed Model|Feature store, both should have their update() calls consolidated with their respective addFeature|Model functions

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/rest/ManagedFeatureStore.java (hunk 1 above)
Comment (jdorando) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/rest/ManagedFeatureStore.java]: We can have 1, but this seems like more the Java style for doing the default param like in C++

**Comment by jdorando** [comment]
code review

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/rest/ManagedFeatureStore.java [hunk 4]
```diff
@@ -113,13 +107,16 @@ public Object applyUpdatesToManagedData(Object updates) {
...
+        final String featureStore = (String) u.get(FEATURE_STORE_NAME_KEY);
+        addFeature(u, featureStore);
       }
     }
 
     if (updates instanceof Map) {
       // a unique feature
-      addFeature((Map<String,Object>) updates);
+      Map<String,Object> updatesMap = (Map<String,Object>) updates;
+      final String featureStore = (String) updatesMap.get(FEATURE_STORE_NAME_KEY);
```
Comment (mnilsson23) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/rest/ManagedFeatureStore.java]: I was thinking this lookup could just be done by the addFeature function

**Pull request merged**